fpga_interchange/arch.cc

```diff
@@ -841,10 +841,6 @@ static void prepare_sites_for_routing(Context *ctx)
         }
     }
 
-    // Clear the site routing cache. This is because routing at this stage is done with the extra constraint of blocked
-    // pins to ensure a routeable pin choice.
-    ctx->site_routing_cache.clear();
```

Comment:
I'm concerned this is going to break UltraScale+ IO (which is where this was added, due to the blocked pins situation), ideally this should be investigated rather than worked around.

Reply:
Indeed, I'll try to get a smaller and reproducible case of the failure when the cache clearing is enabled.

Thinking out loud, it might as well be a bug in the cache, such that some placements are allowed based on a cache that does not exactly represent the real site status.

Reply:
@gatecat After some debugging iterations, I have discovered that there is a possible issue with the routing cache which still needs to be further investigated.

Basically, it seems that during the site routing binding, there could be some cache entries belonging to other sites that produce the effect of having fewer solutions when trying to solve routing for these other sites, making it impossible, in some cases, to find a feasible site routing.

I need to experiment a bit more on the choice of the routing cache key, which may solve the problem, but for now, a short term solution is to disable the usage of the site routing cache when binding the various site routings prior to global routing.